kong/runloop/handler.lua

```diff
@@ -1041,14 +1042,27 @@ return {
 
       local trusted_ip = kong.ip.is_trusted(realip_remote_addr)
       if trusted_ip then
-        forwarded_proto = var.http_x_forwarded_proto or scheme
-        forwarded_host  = var.http_x_forwarded_host  or host
-        forwarded_port  = var.http_x_forwarded_port  or port
+        forwarded_proto  = var.http_x_forwarded_proto  or scheme
+        forwarded_host   = var.http_x_forwarded_host   or host
+        forwarded_port   = var.http_x_forwarded_port   or port
+        forwarded_prefix = var.http_x_forwarded_prefix
 
       else
-        forwarded_proto = scheme
-        forwarded_host  = host
-        forwarded_port  = port
+        forwarded_proto  = scheme
+        forwarded_host   = host
+        forwarded_port   = port
+      end
+
+      if not forwarded_prefix then
+        forwarded_prefix = var.request_uri
+        local p = find(forwarded_prefix, "?", 2, true)
```

Comment:
This removal of the query seems to be done here but not on the pdk function. Shouldn't they behave the same way?

Reply:
PDK also does it, see `get_forwarded_path`.

Reply:
What about introducing a new method to obtain the request path (`get_request_path`)? That would make the code structure similar to the corresponding code in the PDK.

Reply:
@erikgb, I was referring the existing `kong.request.get_path()` that this method calls.

Reply:
@bungle : I know, and that is why I am suggesting to introduce a corresponding method on the handler - to avoid any misunderstanding. I agree with your response to @kikito comment; the handler and the PDK function seems to do the same thing.